### DISCUSSION TITLE
What are the steps to use node modules in the render process (nodeIntegration: true)?

### DISCUSSION THREAD

**Discussion by MiniSuperDev** [alex8088/electron-vite#290] [Q&A]
I know it has security risks, but for now I want to quickly develop the UI for several scripts that I have in node, it is an app that I will only use for now as a proof of concept, and for now it is not worth using IPC.
Related discussion: https://github.com/reZach/secure-electron-template/issues/110#issuecomment-1120634339

I'm using  the react-ts template

https://github.com/alex8088/quick-start/blob/169c4b226b0314814ac066278441f2b85b12e3af/packages/create-electron/template/react-ts/src/main/index.ts#L14-L16

And I have added this:

```diff
    webPreferences: {
      preload: join(__dirname, '../preload/index.js'),
      sandbox: false,
+     contextIsolation: false,
+     nodeIntegration: true
    }
```

For example when use:
```ts
import fetch from 'node-fetch'
import fs from 'fs'
```
I have this errors

```sh 
Module "buffer" has been externalized for browser compatibility. Cannot access "buffer.Blob" in client code. See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.

Module "node:fs" has been externalized for browser compatibility. Cannot access "node:fs.promises" in client code. See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
```

I see you have this in the docs
https://github.com/alex8088/electron-vite-docs/blob/ad802387fec6466fce144d813a85eb7284541d45/packages/en-US/docs/guide/dev.md#nodeintegration

but I honestly didn't quite understand what to do.

I would appreciate if you can give me more details on how to do it.

Thank you so much

**Comment by alex8088** [comment]
https://electron-vite.org/guide/dev.html#nodeintegration
